code/weapon/weapons.cpp

```diff
@@ -4913,6 +4952,28 @@ void weapon_process_post(object * obj, float frame_time)
 
 	wip = &Weapon_info[wp->weapon_info_index];
 
+	if (wip->wi_flags[Weapon::Info_Flags::Spawn]) {
+		for (int i = 0; i < wip->num_spawn_weapons_defined; i++) {
+			if (wip->spawn_info[i].spawn_interval > 0.f) {
+				// continuously advance next_spawn_time and spawn weapons until the time goes into the future
+				// incase we've passed multiple spawn events in the frame
+				while (timestamp_elapsed(wp->next_spawn_time[i])) {
```

Comment:
This scares me a bit. Handled properly it may be okay, but generally you never want `timestamp_elapsed()` in the expression of a `while()` loop. It would be much safer to use a `for()` loop with `wip->maximum_children_spawned` and then check the timestamp in that. That way you've got a guaranteed breakout point.

Even if your specific use case works okay, it sets an example that someone else may copy badly without realizing the dangers of `timestamp_elapsed()`.

Reply:
I wasn't aware of the issues myself, is the worry that `timestamp_elapsed` will take too long or never return false? I think I'll put a 0.1 minimum interval which should be plenty fast even in the most extreme scenarios, so even in this case only very few spawns should be necessary within a frame.

Reply:
`timestamp_elapsed()` isn't really dependable when used like that. It's not slow but there can be conditions where it might result in an endless loop. Your specific usage isn't too bad, but it still worries me.

@Goober5000 Can you take a look at this and tell me if I'm overreacting or not?

Reply:
As long as the timestamp is valid, I can't think of a situation where it would cause an endless loop.  Can you be more specific?

Reply:
Mostly worried about wrapping. It wouldn't be an endless loop, but it would cause a delay until it catches up. That's generally a multi issue though, since it's really only standalone servers that run long enough to wrap the timestamp. If something bad and unrelated happens (memory overwrite, etc.) and the timestamp turns invalid then it might be endless. I'm probably just being overly paranoid though.

@Baezon Your changes here are okay but if you prefer the while() loop instead I guess that would be fine. It's definitely cleaner looking. If it ends up causing a problem later it can be dealt with then.

Reply:
Eh, I might just get rid of it entirely. I was thinking ahead for very small intervals but with a minimum of 0.1 seconds, having to do this more than once is very unlikely for normal gameplay, and you'll have bigger issues if your framerate is 10 or less.